File: step-util/step-rdbms-util/src/main/java/org/garnishtest/modules/generic/db_util/scripts/DbScriptsExecutor.java
package org.garnishtest.modules.generic.db_util.scripts;

import java.sql.Connection;
import java.sql.Types;
import javax.sql.DataSource;
import lombok.NonNull;
import org.flywaydb.core.internal.dbsupport.DbSupport;
import org.flywaydb.core.internal.dbsupport.DbSupportFactory;
import org.flywaydb.core.internal.dbsupport.JdbcTemplate;
import org.flywaydb.core.internal.dbsupport.SqlScript;
import org.garnishtest.modules.generic.db_util.jdbc_transactions.SimpleTransactionTemplate;

import static kotlin.collections.CollectionsKt.joinToString;
import static kotlin.text.StringsKt.lines;

public class DbScriptsExecutor {

    @NonNull
    private final SimpleTransactionTemplate transactionTemplate;

    public DbScriptsExecutor(@NonNull final DataSource dataSource) {
        this.transactionTemplate = new SimpleTransactionTemplate(dataSource);
    }

    public void executeScript(@NonNull final String sqlScript) throws DbScriptsExecutorException {
        this.transactionTemplate.executeInTransaction(connection ->
                DbScriptsExecutor.this.executeScript(connection, sqlScript)
        );
    }

    private void executeScript(@NonNull final Connection connection,
                               @NonNull final String sqlScriptAsString) {
        try {
            final DbSupport dbSupport = createDbSupport(connection);

            final SqlScript sqlScript = new SqlScript(sqlScriptAsString, dbSupport);

            sqlScript.execute(new JdbcTemplate(connection, Types.NULL));
        } catch (final Exception e) {
            String indentScriptLines = joinToString(
                    lines(sqlScriptAsString),
                    "\n",
                    "",
                    "",
                    -1,
                    "...",
                    line -> "\t\t" + line
            );
            throw new DbScriptsExecutorException("failed to execute SQL script\n" + indentScriptLines, e);
        }
    }

    private DbSupport createDbSupport(final Connection connection) {
        final boolean printInfo = true;

        return DbSupportFactory.createDbSupport(connection, printInfo);
    }

}

File: old/step-util/step-rdbms-util/src/main/kotlin/com/testerum/step_rdbms_util/scripts/DbScriptsExecutor.kt
package com.testerum.step_rdbms_util.scripts

import com.testerum.step_rdbms_util.jdbc_transactions.SimpleTransactionTemplate
import org.flywaydb.core.internal.dbsupport.DbSupport
import org.flywaydb.core.internal.dbsupport.DbSupportFactory
import org.flywaydb.core.internal.dbsupport.JdbcTemplate
import org.flywaydb.core.internal.dbsupport.SqlScript
import java.sql.Connection
import java.sql.Types
import javax.sql.DataSource

class DbScriptsExecutor(dataSource: DataSource) {

    private val transactionTemplate: SimpleTransactionTemplate = SimpleTransactionTemplate(dataSource)

    fun executeScript(sqlScriptAsString: String) {
        transactionTemplate.executeInTransaction { connection: Connection ->
            try {
                val dbSupport = createDbSupport(connection)
                val sqlScript = SqlScript(sqlScriptAsString, dbSupport)
                sqlScript.execute(JdbcTemplate(connection, Types.NULL))
            } catch (e: Exception) {
                val indentScriptLines = sqlScriptAsString.lines().joinToString("\n", "", "", -1, "...") { line: String -> "\t\t" + line }
                throw RuntimeException("failed to execute SQL script\n$indentScriptLines", e)
            }
        }
    }


    private fun createDbSupport(connection: Connection): DbSupport {
        val printInfo = true
        return DbSupportFactory.createDbSupport(connection, printInfo)
    }

}
